decorator.js

```diff
@@ -21,11 +21,11 @@ function addClickOutsideListener(Component) {
     },
 
     render: function render() {
-      return React.createElement(Component, objectAssign({
+      return React.createElement(Component, objectAssign({}, this.props, {
```

Comment:
I'm slightly curious why this was set up the way it was, can we do a straight-up passthrough?
```
return React.createElement(Component, objectAssign({ref:'inner'}, this.props));
```
Since the HOC itself should not be doing anything with the properties that got set other than to just pass those on.